program_intent_engagement/apps/core/models.py

```diff
@@ -36,3 +37,43 @@ def get_full_name(self):
 
     def __str__(self):
         return str(self.get_full_name())
+
+
+class ProgramIntent(TimeStampedModel):
+    """
+    Information about the Program Intent
+
+    .. no_pii:
+    """
+
+    user = models.ForeignKey(User, db_index=True, on_delete=models.CASCADE)
+
+    program = models.UUIDField()
+
+    # The specific reason for how intent was measured.
+    reason = models.CharField(max_length=255, db_index=True)
+
+    # Another approach to reason using enum choice field
+    # REASON = (
+    #     ('CC', 'All Courses Complete'),
+    #     ('BP', 'Bundle Purchase'),
+    # )
+    # reason = models.CharField(max_length=2, choices=REASON)
+
+    # The certainty of the program intent for the user.
+    CERTAINTY = (
+        ('Y', 'Certainly Yes'),
+        ('N', 'Certainly No'),
+        ('M', 'Maybe'),
+    )
+
+    certainty = models.CharField(max_length=1, choices=CERTAINTY)
+
+    # When did the event occurred that made this intent.
+    effective_date = models.DateTimeField(null=True)
```

Comment:
not sure if we want this to be nullable - but that's coming from the assumption that every record in this store will be triggered by a specific event

Reply:
Good point, we changed the default to timezone.now

Reply:
I don't think we even want a default - any time someone would use it they're being lazy and losing data